### PR TITLE
[Checklist] Move domain verification nudge to checklist

### LINKED ISSUE #37427: Domain Verification: Remove the nudge from the sidebar and add to checklist
Once a user has completed signup they see the verification nudge in the sidebar:

Screen Shot 2019-11-07 at 3 52 51 PM

In all user tests so far we've seen users distracted by this nudge and take themselves out of the website building flow and lost in the domain verification flow. Verification isn't important until they are ready to launch their site.

Let's start by removing the "Please verify [domain name]" from the sidebar, and replacing it with a checklist item. It should come as the step before launching your homepage:

Screen Shot 2019-11-07 at 4 27 38 PM

### PR DESCRIPTION
**Pull request by adamziel**
#### Changes proposed in this Pull Request

This PR adds some feature-flag gated features:

##### Adds "Verify your domain" checklist item

Zrzut ekranu 2019-11-12 o 14 53 32

This checklist task is incomplete only as long as the site is unlaunched. Completing that item and then adding another domain will re-activate this task. If you launch your site and then add another domain, this task will remain completed.

##### Hide "Verify your domain" nudge from sidebar for unlaunched sites.

68437771-c4569a00-0176-11ea-9c67-7d42b45a9455

That nudge will be only visible for launched sites (when email verification is required).

#### Testing instructions

1. Apply D35374-code
2. Create new account/site without paid domain, confirm the checklist doesn't contain "Verify domain" item and that the same as before
3. Create new account/site with paid domain, confirm the sidebar nudge is gone and that there's a new checklist item "Verify email address for (domain)"
4. Purchase another domain, verify that checklist item is still labeled in a way that makes sense
5. Add another admin user to that site, one that's ineligible to manage the domain. Confirm the sidebar nudge is visible for that user.
6. Verify email address for your domain, confirm the checklist item is now marked as complete.
7. Change email address for your domain, confirm the checklist item is incomplete again. Confirm the sidebar nudge is visible again.
8. Add another user to that site, one that's ineligible to manage the domain. Confirm the sidebar nudge is still in place.
9. Launch the site
10. Change email address for your domain, confirm the checklist item is still complete and that the sidebar nudge is in place instead.
11. Verify the email address, confirm the sidebar nudge is gone

#### Important notes:

* Domain verification is a pre-requisite to "Launch site" task, but it's still possible to launch the site by clicking link on site preview. If this PR proves valuable in user testing, we could sort that pre-requisite in a follow-up diff.
* With this PR we have two very similar tasks: `Verify email address for domain dsadas.blog` and `Confirm your email address`. Ideally we'd only require separate verification when email address assigned to domain is different from email address assigned to account. This is beyond the scope of this PR though.
* Sidebar nudge is only disabled for the first two days. If the user doesn't verify the domain in that timeframe, the nudge is back with a strong message.

#### Additional context

* This will be only open to users participating in usability testing at first.
* If it does well, it would be an A/B test later
* See the following discussion for more context: paObgF-GK-p2

Fixes #37427

### CODE REVIEW

**Comment by adamziel** [comment]
This isn't actually finished or ready to merge, I just wanted to get some eyes on the approach so far

File: client/my-sites/checklist/wpcom-checklist/component.jsx [hunk 1]
```diff
@@ -432,6 +434,41 @@ class WpcomChecklistComponent extends PureComponent {
...
+
+		return (
+			<TaskComponent
+				{ ...baseProps }
+				bannerImageSrc="/calypso/images/stats/tasks/create-tagline.svg"
+				buttonText={ translate( 'Verify' ) }
+				completedButtonText={ translate( 'Change' ) }
+				completedTitle={ translate( 'You verified your domain name' ) }
+				description={ translate(
+					'Email assigned to your domains has to be verified so we can get in touch with you when needed.'
```
Comment (a8ci18n) [inline client/my-sites/checklist/wpcom-checklist/component.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( 'Your email has to be verified so we can get in touch with you when needed.' )` *ES Score: 8*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/domain-verification-nudge-to-checklist)

Comment (daledupreez) [reply client/my-sites/checklist/wpcom-checklist/component.jsx]: We may need to be slightly stronger with the copy here -- we are _legally required_ to confirm the email address to keep the domain available. Without email verification, the domain simply won't work and won't allow the site to load. How about the following?
* `completedTitle` => `You verified the email address for your domain(s)`
* `description` => `The email address for your domains has to be verified to allow your domains to work properly.`

Might it also be worth having `Verify your domains`/`Verify %(domainName)s` be slightly more explicit? e.g. `Verify email address for domains`/`Verify email address for %(domainName)s`. This does confuse _our_ email confirmation, but it is separate/distinct.

File: client/my-sites/checklist/wpcom-checklist/component.jsx [hunk 2]
```diff
@@ -581,8 +618,16 @@ class WpcomChecklistComponent extends PureComponent {
...
 
 	renderSiteLaunchedTask = ( TaskComponent, baseProps, task ) => {
-		const { needsVerification, siteIsUnlaunched, translate } = this.props;
+		const { needsVerification, needsDomainVerification, siteIsUnlaunched, translate } = this.props;
 		const disabled = ! baseProps.completed && needsVerification;
+		let noticeText;
+		if ( disabled ) {
+			if ( needsDomainVerification ) {
+				noticeText = translate( 'Verify your domain before launching your site.' );
+			} else if ( needsDomainVerification ) {
```
Comment (daledupreez) [inline client/my-sites/checklist/wpcom-checklist/component.jsx]: Should this be `needsEmailVerification` or `needsVerification`? As written, this block is unreachable. 😄 (Note that as-is, `needsEmailVerification` is not being pulled out of the props.)

File: client/my-sites/checklist/wpcom-checklist/component.jsx [hunk 3]
```diff
@@ -581,8 +618,16 @@ class WpcomChecklistComponent extends PureComponent {
 	};
 
 	renderSiteLaunchedTask = ( TaskComponent, baseProps, task ) => {
-		const { needsVerification, siteIsUnlaunched, translate } = this.props;
+		const { needsVerification, needsDomainVerification, siteIsUnlaunched, translate } = this.props;
```
Comment (daledupreez) [inline client/my-sites/checklist/wpcom-checklist/component.jsx]: Should we not be importing `needsEmailVerification` instead of `needsVerification`?

File: client/my-sites/domains/components/domain-warnings/index.jsx [hunk 4]
```diff
@@ -633,37 +633,26 @@ export class DomainWarnings extends React.PureComponent {
 					.isAfter()
 		);
 		const severity = isWithinTwoDays ? 'is-info' : 'is-error';
+		if ( severity !== 'is-error' ) {
```
Comment (daledupreez) [inline client/my-sites/domains/components/domain-warnings/index.jsx]: Why not just check for `if ( ! isWithinTwoDays )`? The extra transform seems unnecessary here, especially because we're no longer using `severity` in the code below.

Also, I think it would be well worth adding a comment to say that this step ought to be handled by the checklist, so we only want to flag it when no action has been taken. Otherwise it looks like this is an inexplicable escape clause.

**Review by daledupreez — COMMENTED** [review]
Some comments from me - I am hoping one of the more senior Cobalt members can also weigh in!

**Comment by klimeryk** [comment]
This approach makes a lot of assumptions - starting with "Hides "Verify your domain" nudge from sidebar - **it's clearly distracting users**". It's distracting because otherwise users ignored it and let their domains get suspended. We have track events that should help inform our decisions here, but it does not seem like they were taken into account when proposing this change.

As for the change itself, I don't think putting it in the checklist like this is a good idea - it optimizes for one, very specific use case, but causes many more issues:

 - As you've mentioned, adding another domain causes the checklist to be reactivated. It seems to me like a weird flow where the completed checklist suddenly is not completed again. User could be confused if they need to re-verify the first domain.
 - This approach does not handle very well the very important case where the user edits their domain's contact info and changes the email - which causes the domain to get into the "needs verification" state again. One could assume that at that point, they probably have long finished the checklist and would not think of needing to re-check it again.

If we want to do this right, I'd suggest A/B testing this - we should be able to define clear metrics here (click events on the different verification nudges, for example). Otherwise, we're doing a gamble, with Happiness to handle the possible fallout. And since we're talking about domain suspension, the  full effects would become visible after a week or more (domains already on their way to suspension and new domains not getting verified).

**Comment by adamziel** [comment]
> We have track events that should help inform our decisions here, but it does not seem like they were taken into account when proposing this change.

Do you happen to have any resources handy? I'd love to read more about that! For more context about this change, see the following post and related links: paObgF-GK-p2

> If we want to do this right, I'd suggest A/B testing this - we should be able to define clear metrics here (click events on the different verification nudges, for example).

+1 - As mentioned in the other thread - this will be best served as an A/B test to make sure the domain verification rate doesn't drop: https://github.com/Automattic/wp-calypso/issues/37427

Even more: This will be only open to users participating in usability testing at first. I just updated the description of this PR to reflect some of what I included in this comment.

> As you've mentioned, adding another domain causes the checklist to be reactivated. It seems to me like a weird flow where the completed checklist suddenly is not completed again. User could be confused if they need to re-verify the first domain.

Agreed, and I'd love to get some design feedback before we move on here.

> This approach does not handle very well the very important case where the user edits their domain's contact info and changes the email - which causes the domain to get into the "needs verification" state again. One could assume that at that point, they probably have long finished the checklist and would not think of needing to re-check it again.

Ah! That's a super good point actually. I wonder if the new Customer Home (p9Jlb4-15H-p2) would be a good place to test something like that? Let's definitely get more feedback here.

**Comment by adamziel** [comment]
cc @jancavan @ianstewart @michaeldcain

**Comment by michaeldcain** [comment]
I agree with most of what @klimeryk brings up here. Unlike the second domain upsell nudge, which is purely revenue driven, the domain confirmation nudge serves a legitimate purpose for existing subscribers. If their domain gets suspended, it causes additional support burden and is more likely to lead to churn.

If the problem is that users are getting distracted before editing their sites, I proposed the following in the original issue:

> would it make sense to hide the nudge on the initial customer home (checklist) screen, but show it on others?

That initial view of Customer Home could (should?) be nudge-free. We should still include the confirmation in the Checklist, but given that we don't have a firm grasp on whether or not users return to the Checklist, it feels like the domain confirmation nudge should be shown as it currently is on all subsequent Calypso page views until the user has confirmed their address.

**Comment by ianstewart** [comment]
> One caveat is that verifying all domains, and then adding another domain with different e-mail address will re-activate the checklist item even if it was completed before. 

> It seems to me like a weird flow where the completed checklist suddenly is not completed again. User could be confused if they need to re-verify the first domain.

Suddenly having a checklist item appear again is definitely non-ideal. :D Especially if the checklist disappears post-launch. Would the whole checklist appear again post-launch? How would I know the checklist was back? Would it be up to me to navigate back to the home page, notice the checklist, and understand what it meant.

Agreed on this being something to figure out. :)

> Ah! That's a super good point actually. I wonder if the new Customer Home (p9Jlb4-15H-p2) would be a good place to test something like that? Let's definitely get more feedback here.

The customer home definitely does, however, seem like a good place to make sure that a notification like this is present.

> This approach does not handle very well the very important case where the user edits their domain's contact info and changes the email - which causes the domain to get into the "needs verification" state again. One could assume that at that point, they probably have long finished the checklist and would not think of needing to re-check it again.

What methods do we have now for notifying a user that they need to verify? Do we send emails out?

**Comment by adamziel** [comment]
> What methods do we have now for notifying a user that they need to verify? Do we send emails out?

We do send emails. I've seen one right after purchasing a domain. I'm not sure if there's any reminder after that @klimeryk ?

> The customer home definitely does, however, seem like a good place to make sure that a notification like this is present.

I can see the following solutions here:

1. Leave the current nudge in place, but only start displaying it on day 2 or so.
1. For the first domain/email, present user with a checklist item to complete before launching their site. If any subsequent verification is required, display a nudge on customer home. Seems like good experience for most users and a pretty weird experience for everyone else.
1. Do not change the checklist at all, only move the nudge to Customer Home in some form. Potentially start displaying it on day 2 or so.

cc @ianstewart @apeatling @michaeldcain @jancavan

**Comment by ianstewart** [comment]
> For the first domain/email, present user with a checklist item to complete before launching their site. If any subsequent verification is required, display a nudge on customer home. Seems like good experience for most users and a pretty weird experience for everyone else.

A fourth option that is slightly different than the second option above …

4. If _any_ domains are purchased _before launch_, update the checklist with a checklist item for verification. In other words, while the checklist is still available, use it. If verification is required _after launch_ we could use one of the following options …

4.a. use the existing nudge system in the sidebar after launch as an MVP.
4.b. use the customer home to provide a nudge after launch.

**Comment by adamziel** [comment]
@ianstewart you can launch your site without completing the checklist though:

Zrzut ekranu 2019-11-18 o 16 18 37

**Comment by ianstewart** [comment]
> you can launch your site without completing the checklist though:

Yep. The condition of a nudge appearing being just launch. What I mean is if you haven't launched, keep using the checklist. If you've launched start using something else. And if you launched without verifying your domain via the checklist … then we either show the nudge in the sidebar or on the new customer home.

> I wonder how many people actually do it this way - checklist is not the only place that nudges you to launch the site

Hmm. We should really not show this checklist if you've chosen to launch your site. :D Especially since we have a home screen. (I'll make an issue for this.)

**Comment by ianstewart** [comment]
^^ cc @lcollette and @sixhours on this one as you're working on the checklist and customer home.

**Comment by jancavan** [comment]
> use the customer home to provide a nudge after launch.

We need to be consistent with where we place these nudges. We currently have domain renewal reminders that go on the Notifications (bell) panel. Isn't this more fitting to be on there too?

I feel like the Customer Home's intent is for next steps pertaining to site building/growing their site/etc, so if this were to go there, it would be more appropriate for it to be displayed within the context of a task (eg: sharing their sites).

We also need to be consistent with the colors we use for these nudges - green (success), yellow (warning), red (error/something that requires immediate attention), and gray (informational). I don't even know what the hot pink notification means, it's not a typical color you would use.

**Comment by adamziel** [comment]
I updated this PR a little bit. At the moment:

**The checklist item will only be active for Unlaunched sites with unverified domain(s)**

**The nudge is only going to be visible for:**
1. Launched sites with unverified domain...
2. ...Or Unlaunched sites with unverified domains after 2 days

Reviews/further comments highly appreciated!

File: client/my-sites/current-site/domain-warnings.jsx [hunk 5]
```diff
@@ -17,6 +17,7 @@ import { getSelectedSite, getSelectedSiteId } from 'state/ui/selectors';
 import { isJetpackSite } from 'state/sites/selectors';
 import isSiteAutomatedTransfer from 'state/selectors/is-site-automated-transfer';
 import QuerySiteDomains from 'components/data/query-site-domains';
+import isUnlaunchedSite from '../../state/selectors/is-unlaunched-site';
```
Comment (michaeldcain) [inline client/my-sites/current-site/domain-warnings.jsx]: `from 'state/selectors/is-unlaunched-site';`

File: client/my-sites/checklist/wpcom-checklist/component.jsx [hunk 6]
```diff
@@ -1026,7 +1084,9 @@ export default connect(
 			taskUrls,
 			taskList,
 			userEmail: ( user && user.email ) || '',
-			needsVerification: ! isCurrentUserEmailVerified( state ),
+			needsEmailVerification,
```
Comment (michaeldcain) [inline client/my-sites/checklist/wpcom-checklist/component.jsx]: Is there a reason you moved the selector for `needsEmailVerification`?

File: client/my-sites/checklist/wpcom-checklist/component.jsx [hunk 7]
```diff
@@ -581,8 +618,24 @@ class WpcomChecklistComponent extends PureComponent {
...
+			needsDomainVerification,
+			siteIsUnlaunched,
+			translate,
+		} = this.props;
 		const disabled = ! baseProps.completed && needsVerification;
+		let noticeText;
+		if ( disabled ) {
+			if ( needsDomainVerification ) {
+				noticeText = translate(
+					'Verify email address for your domain before launching your site.'
```
Comment (michaeldcain) [inline client/my-sites/checklist/wpcom-checklist/component.jsx]: I think this should be "Verify the email address for your domain before launching your site."

File: client/my-sites/checklist/wpcom-checklist/component.jsx [hunk 8]
```diff
@@ -432,6 +434,41 @@ class WpcomChecklistComponent extends PureComponent {
...
+							? domainManagementEdit( siteSlug, task.unverifiedDomains[ 0 ] )
+							: domainManagementList( siteSlug ),
+				} ) }
+				onDismiss={ this.handleTaskDismiss( task.id ) }
+				title={
+					task.unverifiedDomains.length === 1
+						? translate( 'Verify email address for domain %(domainName)s', {
+								args: { domainName: task.unverifiedDomains[ 0 ] },
+						  } )
+						: translate( 'Verify email address for domains' )
```
Comment (michaeldcain) [inline client/my-sites/checklist/wpcom-checklist/component.jsx]: I think this should be "Verify the email address for your domains"

File: client/my-sites/checklist/wpcom-checklist/component.jsx [hunk 9]
```diff
@@ -432,6 +434,41 @@ class WpcomChecklistComponent extends PureComponent {
...
+					tourId: 'checklistSiteTagline',
+					url:
+						task.unverifiedDomains.length === 1
+							? domainManagementEdit( siteSlug, task.unverifiedDomains[ 0 ] )
+							: domainManagementList( siteSlug ),
+				} ) }
+				onDismiss={ this.handleTaskDismiss( task.id ) }
+				title={
+					task.unverifiedDomains.length === 1
+						? translate( 'Verify email address for domain %(domainName)s', {
```
Comment (michaeldcain) [inline client/my-sites/checklist/wpcom-checklist/component.jsx]: I think this should be "Verify the email address for %(domainName)s"

File: client/my-sites/checklist/wpcom-checklist/component.jsx [hunk 10]
```diff
@@ -432,6 +434,41 @@ class WpcomChecklistComponent extends PureComponent {
...
+
+		return (
+			<TaskComponent
+				{ ...baseProps }
+				bannerImageSrc="/calypso/images/stats/tasks/create-tagline.svg"
+				buttonText={ translate( 'Verify' ) }
+				completedButtonText={ translate( 'Change' ) }
+				completedTitle={ translate( 'You verified the email address for your domain' ) }
+				description={ translate(
+					'The email address for your domain has to be verified to allow your domains to work properly.'
```
Comment (michaeldcain) [inline client/my-sites/checklist/wpcom-checklist/component.jsx]: We could use the following from the domains screen to prevent needing to translate new strings:

> We need to check your contact information to make sure you can be reached. Please verify your details using the email we sent you, or your domain will stop working.

File: client/state/data-layer/wpcom/checklist/index.js [hunk 11]
```diff
@@ -23,7 +24,11 @@ const fromApi = payload => get( payload, 'body', payload );
 export const fetchChecklist = action =>
 	http(
 		{
-			path: `/sites/${ action.siteId }/checklist`,
+			path:
+				`/sites/${ action.siteId }/checklist` +
+				( config.isEnabled( 'experience/domain-verification-in-checklist' )
```
[the remark this answers is not in the document]
Comment (adamziel) [reply client/state/data-layer/wpcom/checklist/index.js]: The backend only returns the task if the query param is present, so this is a way of saying "I have feature flag enabled". There are other ways of doing that too but this should be fine.

**Comment by apeatling** [comment]
@adamziel I tested this locally with a brand new user in a incognito window. In the checklist the email address confirmation was already checked?

Screen Shot 2019-11-19 at 12 31 30 PM

**Comment by p-jackson** [comment]
The logic for hiding the sidebar nudge during the first two days looks like it's the wrong way round. I see the nudge in the sidebar immediately after finishing signup.
Will keep reviewing and push fix if I get there before anyone else.

File: client/state/data-layer/wpcom/checklist/index.js (hunk 11 above)
[the remark this answers is not in the document]
Comment (p-jackson) [reply client/state/data-layer/wpcom/checklist/index.js]: I agree, it seems like we should change query below to:

```js
query: {
  http_envelope: 1,
  with_domain_verification: config.isEnabled( 'experience/domain-verification-in-checklist' ) ? 1 : 0,
}
```

I might be missing why that might not work though. Works the way it is though, I don't mind sending it to user testing as is.

**Comment by p-jackson** [comment]
> The logic for hiding the sidebar nudge during the first two days looks like it's the wrong way round

I've fixed the issue above, but am having trouble actually testing the domain verification step. I'm not recieving any emails to do with domains, even after clicking re-send. Tried with store sandboxed and unsandboxed.

Maybe an issue with D35374-code or maybe something's just wrong with my setup.

**Comment by p-jackson** [comment]
Found a couple of issues.

#### Changing domain contact email

> 7. Change email address for your domain, confirm the checklist item is still complete. Confirm the sidebar nudge is visible again.

This didn't happen for me. After my domain contact email (but before verifying it) the domain verification checklist item got unchecked, and the sidebar nudge did not appear. It wasn't that confusing tbh, but not the originally expected behaviour.

#### Verifying two domains at the same time

What looked like domain verification emails running very slow may actually be an issue with verifying two domains at the same time (so not related to this PR). On step 6 of the testing instructions, after following the verification link in the email the message said both domains had been verified. But looking in the calypso UI it appeared that only one had been verified. I could press the "Send Again" button but it said one was already in the queue, I think because the backend already thinks the second domain has been verified.

---

I'm not too familiar with domains but these seem like issues we should try to understand before deploying to everyone; and I'm not confident enough to say it wouldn't impact the upcoming user testing session.

File: client/my-sites/checklist/wpcom-checklist/component.jsx [hunk 12]
```diff
@@ -432,6 +434,41 @@ class WpcomChecklistComponent extends PureComponent {
 		);
 	};
 
+	renderDomainVerifiedTask = ( TaskComponent, baseProps, task ) => {
+		const { translate, siteSlug } = this.props;
+
+		return (
+			<TaskComponent
+				{ ...baseProps }
+				bannerImageSrc="/calypso/images/stats/tasks/create-tagline.svg"
```
Comment (andrewserong) [inline client/my-sites/checklist/wpcom-checklist/component.jsx]: This is just a tiny thing because I don't think this banner image ends up being shown anywhere, but should this either have its own `svg`, or otherwise be blank?

**Comment by daledupreez** [comment]
@p-jackson, the email verification process for domains only needs verification to be performed once for a given email address, so one verification click/acceptance will verify that email address for all domains that have that address as the domain contact. For testing purposes here, it's probably worth adding a `+suffix` to your email address to make it not match a previously verified email address.

File: client/state/data-layer/wpcom/checklist/index.js (hunk 11 above)
[the remark this answers is not in the document]
Comment (adamziel) [reply client/state/data-layer/wpcom/checklist/index.js]: Ah I missed the point the first time - thank you for clarifying

File: client/my-sites/domains/components/domain-warnings/index.jsx [hunk 13]
```diff
@@ -637,7 +637,7 @@ export class DomainWarnings extends React.PureComponent {
 		if (
 			config.isEnabled( 'experience/domain-verification-in-checklist' ) &&
 			this.props.siteIsUnlaunched &&
-			! isWithinTwoDays
```
Comment (adamziel) [inline client/my-sites/domains/components/domain-warnings/index.jsx]: Good catch!

**Comment by adamziel** [comment]
> This didn't happen for me. After my domain contact email (but before verifying it) the domain verification checklist item got unchecked, and the sidebar nudge did not appear. It wasn't that confusing tbh, but not the originally expected behaviour.

Actually that's the expected behavior before launching the site - I just updated the test plan to make it clearer

**Comment by adamziel** [comment]
@apeatling that's a super good spot - I checked with Cobalt and it seems like registrar is already quite familiar with the e-mail you provided for that domain and just assumed it's verified by now. Not displaying the checklist task for these "verified by default" cases would be unnecessarily difficult so I think that the way to go would be a copy change in that completed item.

**Comment by adamziel** [comment]
Actually I'm going to merge this one for the session later today since it's behind the feature flag. Let's figure out all the rough edges in a follow-up PR

File: test/e2e/specs/wp-signup-spec.js [hunk 14]
```diff
@@ -1763,7 +1763,9 @@ describe( `[${ host }] Sign Up  (${ screenSize }, ${ locale })`, function() {
 		} );
 	} );
 
-	describe( 'Passwordless signup @parallel', function() {
+	// Disable test while Passwordless functionality is completely switched off
+	// https://github.com/Automattic/wp-calypso/pull/37054
+	describe.skip( 'Passwordless signup @parallel', function() {
```
Comment (klimeryk) [inline test/e2e/specs/wp-signup-spec.js]: Seems unrelated?

**Comment by klimeryk** [comment]
In the spirit of moving fast and avoiding confusions, I'd strongly recommend in the future to pair with a Cobalt dev instead of trying to figure out on your own the rules for domain verification 🙇 We're a helpful bunch, we don't bite ;)

File: test/e2e/specs/wp-signup-spec.js (hunk 14 above)
Comment (adamziel) [reply test/e2e/specs/wp-signup-spec.js]: Yup, it seems like it got there in this commit: https://github.com/Automattic/wp-calypso/pull/37510/commits/469ee000da56853cfa99bb3a4dad54f95f1bff1f

I'm going to remove it

**Comment by adamziel** [comment]
> In the spirit of moving fast and avoiding confusions, I'd strongly recommend in the future to pair with a Cobalt dev instead of trying to figure out on your own the rules for domain verification 🙇 We're a helpful bunch, we don't bite ;)

Good point! In this case I've been reusing the same rules we've had in place for the nudge already - but considering the corner case @apeatling ran into, it would have been totally worth it to have that conversation earlier.

File: test/e2e/specs/wp-signup-spec.js (hunk 14 above)
Comment (adamziel) [reply test/e2e/specs/wp-signup-spec.js]: cc @p-jackson

Comment (p-jackson) [reply test/e2e/specs/wp-signup-spec.js]: Yes I did it to get CI passing. Rebasing the branch would have been the less confusing thing to do.

**Comment by p-jackson** [comment]
> the email verification process for domains only needs verification to be performed once for a given email address, so one verification click/acceptance will verify that email address for all domains that have that address as the domain contact.

Thanks for the explanation @daledupreez. I'm going to test again using my new knowledge and will reach out to Cobalt if something still seems fishy.
